lib/puppet/type/quagga_interface.rb

```diff
@@ -0,0 +1,194 @@
+Puppet::Type.newtype(:quagga_interface) do
+  @doc = 'This type provides the capabilities to manage Quagga interface parameters'
+
+  feature :enableable, "The provider can enable and disable the interface", :methods => [:disable, :enable, :enabled?]
+
+  newproperty(:enable, :required_features => :enableable) do
+     desc 'Whether the interface should be enabled or not'
+
+     newvalue(:true) do
+       provider.enable
+     end
+
+     newvalue(:false) do
+       provider.disable
+     end
+
+     def retrieve
+       provider.enabled?
+     end
+  end
+
+  ensurable do
```

Comment:
@bibigon812 - hm, I guess it will not actually remove the virtual device associated to the physical nic itself but it will effectively shut it down. However, let's say you define a new logical interface, e.g. tun0, and you want to remove it after, you still need a way of doing that, that is where ensure => absent comes in.

I guess 'ensure => absent' should be used with care. If one wants to enable or disable (shutdown) the interface, enable => true/false would do the trick.